scheduler/mesos/mesos.go

```diff
@@ -222,56 +235,71 @@ func (m *mesosScheduler) getMesosMaster() ([]string, string, error) {
 	return masterHosts, protocol, nil
 }
 
-func (m *mesosScheduler) getMesosTask(taskId string) (mesosTask, string, string, error) {
-	var state mesosState
-	var masterErr error
-	if masterHosts, protocol, err := m.getMesosMaster(); err == nil {
-		for _, host := range masterHosts {
-			if resp, err := http.Get(protocol + "://" + host + "/state.json"); err == nil {
-				defer resp.Body.Close()
-				if err := json.NewDecoder(resp.Body).Decode(&state); err == nil {
-					if state.Pid == state.Leader {
-						masterErr = nil
-						break
-					}
-				} else {
-					masterErr = err
+func (m *mesosScheduler) getMesosTask(taskID string) (mesosTask, string, string, error) {
```

Comment:
Regarding the 4 returns, is this following a prior design or is this being introduced with your changes? I personally would rather see the frist three in a `struct` and keep returns to 2, if at all possible.

Reply:
I agree with you but this is prior design. I cannot change it without changing other parts too.